src/plugins/github/parseReferences.js

```diff
@@ -98,6 +99,47 @@ function findGithubUrlReferences(textBlock: string): ParsedReference[] {
   }));
 }
 
+function findCommitReferences(textBlock: string): ParsedReference[] {
+  const hashReferences = findAllMatches(
+    /(?:\W|^)([a-fA-F0-9]{40})(?=\W|$)/gm,
+    textBlock
+  ).map((x) => ({
+    refType: "BASIC",
+    ref: x[1].toLowerCase(),
+  }));
+  const commitUrlRegex = new RegExp(
+    "" +
+      /(?:\W|^)/.source +
+      "(" +
+      /https?:\/\/github.com\//.source +
+      `(?:${githubOwnerPattern})` +
+      /\//.source +
+      `(?:${githubRepoPattern})` +
+      /\/commit\/([a-z0-9]{40})/.source +
```

Comment:
`[a-z0-9]` should be `[a-fA-F0-9]`.

Do you want to add a test case with uppercase in the URL?